xonsh/parsers/completion_context.py

```diff
@@ -68,23 +83,44 @@ def raw_prefix(self):
 
 
 class PythonContext(NamedTuple):
+    """
+    The object containing the current python code completion context.
+    """
+
     multiline_code: str
+    """The multi-line python code"""
     cursor_index: int
+    """The cursor's index in the multiline code"""
     is_sub_expression: bool = False
-    ctx: Optional[Dict[str, Any]] = None  # Objects in the current execution context
+    """Whether this is a sub expression (``@(...)``)"""
+    ctx: Optional[Dict[str, Any]] = None
+    """Objects in the current execution context"""
 
     def __repr__(self):
         # don't show ctx since it might be huge
-        return f"PythonContext({self.multiline_code}, {self.cursor_index}, is_sub_expression={self.is_sub_expression})"
+        return f"PythonContext({self.multiline_code!r}, {self.cursor_index}, is_sub_expression={self.is_sub_expression})"
 
     @property
     def prefix(self):
+        """The code from the start to the cursor (may be multiline)"""
         return self.multiline_code[: self.cursor_index]
 
 
 class CompletionContext(NamedTuple):
+    """
+    The object containing the current completion context.
+    """
+
     command: Optional[CommandContext] = None
+    """
+    The current command.
+    This will be ``None`` when we can't be completing a command, e.g. ``echo @(<TAB>``.
+    """
```

Comment:
I think you wanted to mention the reverse

Reply:
Nope - in the example we aren't completing a command, for example `echo @(ls -l)` won't work